ibmcloudshellv1/ibm_cloud_shell_v1_integration_test.go

```diff
@@ -0,0 +1,173 @@
+			}
+
+			os.Setenv("IBM_CREDENTIALS_FILE", externalConfigFile)
+			config, err = core.GetServiceProperties(ibmcloudshellv1.DefaultServiceName)
+			if err != nil {
+				Skip("Error loading service properties, skipping tests: " + err.Error())
+			}
+			serviceURL = config["URL"]
+			if serviceURL == "" {
+				Skip("Unable to load service URL configuration property, skipping tests")
+			}
+
+			fmt.Printf("Service URL: %s\n", serviceURL)
+			shouldSkipTest = func() {}
+		})
+	})
+
+	Describe(`Client initialization`, func() {
+		BeforeEach(func() {
+			shouldSkipTest()
+		})
+		It("Successfully construct the service client instance", func() {
+
+			ibmCloudShellServiceOptions := &ibmcloudshellv1.IbmCloudShellV1Options{}
+
+			ibmCloudShellService, err = ibmcloudshellv1.NewIbmCloudShellV1UsingExternalConfig(ibmCloudShellServiceOptions)
+
+			Expect(err).To(BeNil())
+			Expect(ibmCloudShellService).ToNot(BeNil())
+			Expect(ibmCloudShellService.Service.Options.URL).To(Equal(serviceURL))
+		})
+	})
+
+	Describe(`GetAccountSettingsByID - Get account settings`, func() {
+		BeforeEach(func() {
+			shouldSkipTest()
+		})
+		It(`GetAccountSettingsByID(getAccountSettingsByIdOptions *GetAccountSettingsByIdOptions)`, func() {
+
+			getAccountSettingsByIdOptions := &ibmcloudshellv1.GetAccountSettingsByIdOptions{
+				AccountID: core.StringPtr("12345678-abcd-1a2b-a1b2-1234567890ab"),
+			}
+
+			accountSettings, response, err := ibmCloudShellService.GetAccountSettingsByID(getAccountSettingsByIdOptions)
+
+			Expect(err).To(BeNil())
+			Expect(response.StatusCode).To(Equal(200))
+			Expect(accountSettings).ToNot(BeNil())
+
+		})
+	})
+
+	Describe(`UpdateAccountSettingsByID - Update account settings`, func() {
+		BeforeEach(func() {
+			shouldSkipTest()
+		})
+		It(`UpdateAccountSettingsByID(updateAccountSettingsByIdOptions *UpdateAccountSettingsByIdOptions)`, func() {
+
+			featureModel := []ibmcloudshellv1.Feature{
+				{
+					Enabled: core.BoolPtr(true),
+					Key:     core.StringPtr("server.file_manager"),
+				},
+				{
+					Enabled: core.BoolPtr(true),
+					Key:     core.StringPtr("server.web_preview"),
+				},
+			}
+
+			regionSettingModel := []ibmcloudshellv1.RegionSetting{
+				{
+					Enabled: core.BoolPtr(true),
+					Key:     core.StringPtr("eu-de"),
+				},
+				{
+					Enabled: core.BoolPtr(true),
+					Key:     core.StringPtr("jp-tok"),
+				},
+				{
+					Enabled: core.BoolPtr(true),
+					Key:     core.StringPtr("us-south"),
+				},
+			}
+
+			accountID := "12345678-abcd-1a2b-a1b2-1234567890ab"
+			updateAccountSettingsByIdOptions := &ibmcloudshellv1.UpdateAccountSettingsByIdOptions{
+				AccountID:                   core.StringPtr(accountID),
+				NewID:                       core.StringPtr("ac" + accountID),
+				NewRev:                      core.StringPtr("130" + accountID),
+				NewAccountID:                core.StringPtr(accountID),
+				NewCreatedAt:                core.Int64Ptr(int64(1600079615)),
```

Comment:
It's odd that the service implementation would allow you to set fields like "created_at", "created_by", "updated_at", "updated_by".  These types of fields are usually populated by the server  and treated as "readonly" from the client's perspective.   I think you should consider setting their "readonly" field to true in the API definition if this is the case.
If you set them to readonly, then those fields will not appear in the options model struct, and you will simply be unable to set them when invoking the Update operation.   Those fields would still be a part of the AccountSettings struct and you can read them after calling the Get operation.

Reply:
And remember to regenerate the examples so they won't have these fields.

Reply:
Done